### ISSUE TITLE
Create custom error pages

### ISSUE THREAD

**Issue by AlexxNica** [pln-planning-tools/MapLight#51]
We should have custom error pages.
- https://nextjs.org/docs/advanced-features/custom-error-page

**Comment by SgtPooki** [comment]
We need to nail down our edge-cases and document exactly what errors we need to handle, and how they should be handled 

### Known errors
* Invalid github issue URL provided
* Invalid schema definition in github issue
* ?

### Questions
1. does one error page suffice or do we need unique error pages for different errors?
    * To know that, we need to know what errors users may run across, and what UX we want them to have when they run into them

**Comment by juliaxbow** [comment]
Re: Invalid github metadata, Reid and I talked about including an error summary.  Let me know what you think, @wilkyr31d and @SgtPooki let me know how feasible you think this is.

Included dark/light mode below. Also included two options for presenting the errors: single column or two column.  A row with github link and description of error makes sense if the error description is minimal. If the error description will be longer, I'd switch to the single column view. Can play around with wording, too, but should align on standard error messages.


Dark Mode GitHub Metadata Error Summary (Rows)
<img width="647" alt="Dark Mode GitHub Metadata Error Summary Rows" src="https://user-images.githubusercontent.com/58958327/201447886-5913bd77-eff7-4454-b5d7-e3a3c0592693.png">

Dark Mode GitHub Metadata Error Summary (Single Column)
<img width="655" alt="Dark Mode GitHub Metadata Error Summary Columns" src="https://user-images.githubusercontent.com/58958327/201447884-b753f6ac-b4f3-47a8-b18b-496fc8b87f49.png">

Light Mode GitHub Metadata Error Summary (Single Column)
<img width="654" alt="Light Mode GitHub Metadata Error Summary Columns" src="https://user-images.githubusercontent.com/58958327/201447887-ff4681f2-df7e-4e6d-9d71-d7a9aed45e70.png">

Light Mode GitHub Metadata Error Summary (Rows)
<img width="654" alt="Light Mode GitHub Metadata Error Summary Rows" src="https://user-images.githubusercontent.com/58958327/201447888-13514a83-b686-47db-9bb6-f8b67356795f.png">

**Comment by wilkyr31d** [comment]
@juliaxbow - generally speaking, I think the representation looks great. Not sure on this point: should the error list start out collapsed with only an indicator that there are errors visible?

**Comment by juliaxbow** [comment]
Agreed on allowing users to collapse the error log (I suggest we allow collapsing and remove the ability for users to close out the error log). Convo happening over slack.

I also updated this issue to include a link to error mapping in Notion

**Comment by SgtPooki** [comment]
Do we want to consider an error where an issue's ETA is earlier than one of it's children?  I think adding that at some point could be really useful to debug complex roadmaps.

**Comment by tinytb** [comment]
Minor suggestion: For the "Issues with roadmap" dropdown, instead of only the chevron on the right [circled in red] being clickable, I expected the entire rectangle [outlined in red] to be clickable.

<img width="1271" alt="Screen Shot 2022-11-17 at 10 01 27 PM" src="https://user-images.githubusercontent.com/5957298/202632061-2ea8566f-5b9f-494b-af68-5c2193d5b9e0.png">

**Comment by SgtPooki** [comment]
Actions items left to close this out:

- [ ] Make "Issues with roadmap" entire header clickable to expand/collapse
- [ ] Ensure an error shows when ETA is missing, but is required
- [ ] Ensure error does not show if ETA is missing for currently viewed root issue
- [ ] Ensure error does not show if ETA is missing (on detail view) for groupParent issue
